dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/query/TeiFields.java

```diff
@@ -149,4 +169,18 @@ public static Set<GridHeader> getGridHeaders( final TeiQueryParams teiQueryParam
 
         return headers;
     }
+
+    /**
+     * Static element is term for the parameter used directly as a database
+     * table column name (example: OU, uidlevel1, ..) Dynamic elements are for
+     * example columns with uid strings
+     *
+     * @param p AnalyticsSortingParas
+     * @return boolean
+     */
+    private static boolean isDynamicElement( DimensionIdentifier<Program, ProgramStage, DimensionParam> p )
+    {
+        return (p.hasProgram() || p.hasProgramStage()) &&
```

Comment:
this is fine, however no dimension exists having only the programStage (so if program is null, then also programStage should be null)